src/main/java/org/cactoos/time/DateOf.java

```diff
@@ -64,13 +68,29 @@ public DateOf(final CharSequence date, final String format) {
      * @param formatter The formatter to use.
      */
     public DateOf(final CharSequence date, final DateTimeFormatter formatter) {
+        this(formatter.parse(date));
+    }
+
+    /**
+     * Getting a date from a temporalAccesor.
+     * @param temporal The temporal accessor to use.
+     */
+    public DateOf(final TemporalAccessor temporal) {
         this.parsed = new UncheckedScalar<>(
-            () -> Date.from(
-                LocalDateTime.from(
-                    formatter.parse(date)
-                ).toInstant(ZoneOffset.UTC)
-            )
-        );
+            new Ternary<>(
+                () -> temporal.isSupported(ChronoField.HOUR_OF_DAY),
+                () -> Date.from(
+                    LocalDateTime
+                        .from(temporal)
+                        .toInstant(ZoneOffset.UTC)
```

Comment:
@strokyl this temporal accessor may already come with its own zone offset

Reply:
Well seen, I added test for datetime with timezone and took the timezone of the temporal if it has any